server/middlewares/passport.js

```diff
@@ -0,0 +1,54 @@
+const CONFIG = require('config')
+const __ = CONFIG.universalPath
+const _ = __.require('lib', 'utils')
+const session = require('express-session')
+const passport = require('passport')
+const User = __.require('controllers', '/users/lib/users')
+const gitlabStrategy = require('./strategies/gitlab')
+
+// Passport serialize and deserialize user instances to and from the session
+passport.serializeUser(function (user, done) {
+  const id = user.id || user._id
+  _.success(id, 'serializeUser')
+  done(null, id)
+})
+
+passport.deserializeUser(function (id, done) {
+  User.byId(id)
+  .then(function (user) {
+    _.success(user, 'deserializeUser')
+    done(null, user)
+  })
+  .catch(function (err) {
+    _.error(err, 'deserializeUser err')
+    done(err)
+  })
+})
+
+// assign to passport current session a default strategy
+passport.use('gitlab', gitlabStrategy.auth)
+
+const redirectionParams = {
+  failureRedirect: '/',
+  successReturnToOrRedirect: CONFIG.get('auth.clientEndpoint'),
```

Comment:
If we want generalised clients also other than the current editor, we may want to send the users back to where they came from.